src/commands/up.rs

```diff
@@ -387,25 +413,46 @@ impl UpCommand {
 }
 
 #[cfg(windows)]
-fn set_kill_on_ctrl_c(trigger_processes: &Vec<tokio::process::Child>) -> Result<(), anyhow::Error> {
+fn set_kill_on_ctrl_c(_pids: &[usize]) -> Result<(), anyhow::Error> {
     Ok(())
 }
 
 #[cfg(not(windows))]
-fn set_kill_on_ctrl_c(trigger_processes: &[tokio::process::Child]) -> Result<(), anyhow::Error> {
+fn set_kill_on_ctrl_c(pids: &[nix::unistd::Pid]) -> Result<(), anyhow::Error> {
+    let pids = pids.to_owned();
+    ctrlc::set_handler(move || {
+        kill_them_all_god_will_know_his_own(&pids);
+    })?;
+    Ok(())
+}
+
+#[cfg(windows)]
+fn get_pids(_trigger_processes: &[tokio::process::Child]) -> Vec<usize> {
+    vec![]
+}
+
+#[cfg(not(windows))]
+fn get_pids(trigger_processes: &[tokio::process::Child]) -> Vec<nix::unistd::Pid> {
+    use itertools::Itertools;
     // https://github.com/nix-rust/nix/issues/656
-    let pids = trigger_processes
+    trigger_processes
         .iter()
         .flat_map(|child| child.id().map(|id| nix::unistd::Pid::from_raw(id as i32)))
-        .collect_vec();
-    ctrlc::set_handler(move || {
-        for pid in &pids {
-            if let Err(err) = nix::sys::signal::kill(*pid, nix::sys::signal::SIGTERM) {
-                tracing::warn!("Failed to kill trigger handler process: {:?}", err)
-            }
+        .collect_vec()
+}
+
+#[cfg(windows)]
+fn kill_them_all_god_will_know_his_own(_pids: &[usize]) {}
+
+#[cfg(not(windows))]
+fn kill_them_all_god_will_know_his_own(pids: &[nix::unistd::Pid]) {
+    // https://github.com/nix-rust/nix/issues/656
+    for pid in pids {
+        // println!("killing {pid}");
+        if let Err(err) = nix::sys::signal::kill(*pid, nix::sys::signal::SIGTERM) {
+            tracing::warn!("Failed to kill trigger handler process: {:?}", err)
```

Comment:
I think this could be simplified a bit by passing the `u32`s from `Child::id` into `kill_...` and doing the cast to `Pid`s in there.